index.d.ts

```diff
@@ -1245,6 +1246,22 @@ declare namespace plugins {
    */
   interface router extends Integration {}
 
+  /**
+   * This plugin automatically instruments the
+   * [sharedb](https://github.com/share/sharedb) module.
+   */
+  interface sharedb extends Integration {
+    /**
+     * Hooks to run before spans are finished.
+     */
+    hooks?: {
+      /**
+       * Hook to execute just when the span is created.
+       */
+      receive?: (span?: opentracing.Span, shareDBAgent?: any, shareDBTriggerContext?: any) => any;
```

Comment:
It looks like `agent` and `context` are known concepts of ShareDB, so I'd probably call them just that, without the additional `shareDB` and/or `trigger`. It also seems that every action has an `agent` property, which I'm guessing is the same as the agent passed here, so it seems like it's redundant and only the context is needed.

For the name of the span/hook, my understanding from going through the doc is that the server has both a `receive` and `reply` actions which together consist a request operation. Since the span represents the entire operation, I'd be inclined to name this `request` instead of `receive`, unless you think it makes sense to have individual spans and hooks for both actions.

Reply:
Will make the name changes - however I think that **when** the span is created is just as important as the span itself, so I would propose having two hooks: `receive` and `reply`. While different from say the ExpressJS plugin, this aligns with the conventions ShareDB uses.